samples/Exporters/Console/TestConsoleActivity.cs

```diff
@@ -0,0 +1,95 @@
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using OpenTelemetry.Exporter.Console;
+using OpenTelemetry.Trace.Configuration;
+using OpenTelemetry.Trace.Export;
+
+namespace Samples
+{
+    internal class TestConsoleActivity
+    {
+        internal static object Run(ConsoleActivityOptions options)
+        {
+            // Setup exporter
+            var exporterOptions = new ConsoleActivityExporterOptions
+            {
+                DisplayAsJson = options.DisplayAsJson,
+            };
+            var activityExporter = new ConsoleActivityExporter(exporterOptions);
+
+            // Setup processor
+            var activityProcessor = new SimpleActivityProcessor(activityExporter);
+
+            // Enable OpenTelemetry for the "source" named "MyCompany.MyProduct.MyWebServer".
+            OpenTelemetrySDK.EnableOpenTelemetry("MyCompany.MyProduct.MyWebServer", activityProcessor);
+
+            // Everything above this line is required only in Applications
+            // which decide to use OT.
+
+            // The following is generating activity.
+            // Libraries would simply write the following lines of code.
+            var source = new ActivitySource("MyCompany.MyProduct.MyWebServer");
+
+            // The below commented out line shows more likely code a in real world webserver.
+            // using (var parent = source.StartActivity("HttpIn", ActivityKind.Server, HttpContext.Request.Headers["traceparent"] ))
+            using (var parent = source.StartActivity("HttpIn", ActivityKind.Server))
+            {
+                parent?.AddTag("http.method", "GET");
+                parent?.AddTag("http.host", "MyHostName");
+                if (parent != null)
+                {
+                    parent.DisplayName = "HttpIn DisplayName";
+                }
+
+                try
+                {
+                    // Actual code to achieve the purpose of the library.
+                    // For websebserver example, this would be calling
+                    // user middlware pipeline.
+
+                    // There can be child activities.
+                    // In this example HttpOut is a child of HttpIn.
+                    using (var child = source.StartActivity("HttpOut", ActivityKind.Client))
+                    {
+                        child?.AddTag("http.url", "www.mydependencyapi.com");
+                        try
+                        {
+                            // do actual work.
+
+                            child?.AddEvent(new ActivityEvent("sample activity event."));
+                            child?.AddTag("http.status_code", "200");
+                        }
+                        catch (Exception)
+                        {
+                            child?.AddTag("http.status_code", "500");
+                        }
+                    }
+
+                    parent?.AddTag("http.status_code", "200");
```

Comment:
According to the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/http.md#common-attributes) it is an integer.

Reply:
yup. Tags are currently string,string only. This is a drift from OT spec.

Reply:
I think `http.status_code` should use `parent.SetCustomProperty("http.status_code", 200)` rather than tag?

Reply:
The current adapters use Attributes for storing status_code with ("http.status_code", 200). Attributes support int, but its replacement Activity.Tags only support string.

On top of this, Spans have Status, which don't have a place in Activity yet - the suggestion was to use activity.SetCustomProperty, for the SpanStatus. This was not finalized as well, as there was plans to drop the Status from Span?.

Agree this needs to be finalized - do we need it for this PR or we can do it when more clarity is achieved?
My simplest Exporter is not looking at activity.CustomProperties. Would add it in subsequent PR.

Reply:
discussion in dotnet activity PR: https://github.com/dotnet/designs/pull/98/files#diff-25bba2a21bc64ddeccd374c1c28a0fefR406

Reply:
No need to block this PR. I suggest that we remove this line to avoid examples that are not delivering clarity.

Reply:
`parent?.AddTag("http.status_code", "200");` - This looks fine to me as in old Span model also, we were adding the same as Span.Attributes. And this is in sync with OT conventions. (putting http status code as attribute named "http.status_code".

Whats missing now is the equivalent of Span.Status.
If OT makes Status more generic, and then activity will introduce Activity.Status - then we'll use it.
If OT is not willing make any change to Status, then we'll be forced to use Activity.SetCustomProperty for storing Status. 
This part I am omitting completely for now. 

Let me know if this makes sense.

Reply:
> Tags are currently string,string only. This is a drift from OT spec.

It will be good if exporters (at least the ones on this repo) have consistent behavior in this regard, eg.: by default have every `value` as a `string` and optionally enable a best-effort to convert to some "supported" types by the exporter format.

> ... as there was plans to drop the Status from Span?.

is anybody actively pursuing that?

Reply:
If the discussion here regarding supporting the OT Status type, then SetCustomProperty can be used here. But if we are talking about adding simple HTTP status code,  then I believe using Tags will be more reasonable as it will log a simple code and no further information attached to it (e.g. description).

Reply:
For HTTP the existing semantic conventions are already enough (and even the OTel Status can be derived from that). For others wanting to set span status or equivalent, we need further conventions to do the same.

Reply:
Is there any pending work to support non-string values as activity tags?

Reply:
> Is there any pending work to support non-string values as activity tags?

Yes. Tarek is working on proposals.